integration/kubernetes/vfio.sh

```diff
@@ -0,0 +1,153 @@
+
+SYSCONFIG_FILE="/etc/kata-containers/configuration.toml"
+
+trap cleanup EXIT
+cleanup() {
+	rm -rf "${SYSCONFIG_FILE}"
+	${SCRIPT_DIR}/cleanup_env.sh
+}
+
+setup_configuration_file() {
+	local image_type="$1"
+	local machine_type="$2"
+	local hypervisor="$3"
+	local sandbox_cgroup_only="$4"
+
+	local default_config_file="/usr/share/defaults/kata-containers/configuration.toml"
+	local qemu_config_file="/usr/share/defaults/kata-containers/configuration-qemu.toml"
+	local clh_config_file="/usr/share/defaults/kata-containers/configuration-clh.toml"
+	local image_file="/usr/share/kata-containers/kata-containers.img"
+	local initrd_file="/usr/share/kata-containers/kata-containers-initrd.img"
+
+	mkdir -p $(dirname "${SYSCONFIG_FILE}")
+
+	if [ "${hypervisor}" = "qemu" ]; then
+		config_file="${qemu_config_file}"
+	elif [ "${hypervisor}" = "cloud-hypervisor" ]; then
+		config_file="${clh_config_file}"
+	fi
+
+	if [ -f "${config_file}" ]; then
+		cp -a "${config_file}" "${SYSCONFIG_FILE}"
+	elif [ -f "${default_config_file}" ]; then
+		# Check if path contains the hypervisor name
+		grep "^path" "${default_config_file}" | grep -q "${hypervisor}"
+		cp -a "${default_config_file}" "${SYSCONFIG_FILE}"
+	else
+		die "Error: configuration file for ${hypervisor} doesn't exist"
+	fi
+
+	# machine type applies to configuration.toml and configuration-qemu.toml
+	if [ -n "${machine_type}" ]; then
+		if [ "${hypervisor}" = "qemu" ]; then
+			sed -i 's|^machine_type.*|machine_type = "'${machine_type}'"|g' "${SYSCONFIG_FILE}"
+		else
+			info "Variable machine_type only applies to qemu. It will be ignored"
+		fi
+	fi
+
+	if [ -n "${sandbox_cgroup_only}" ]; then
+		sed -i 's|^sandbox_cgroup_only.*|sandbox_cgroup_only='${sandbox_cgroup_only}'|g' "${SYSCONFIG_FILE}"
+	fi
+
+	# Change to initrd or image depending on user input.
+	# Non-default configs must be changed to specify either initrd or image, image is default.
+	if [ "${image_type}" = "initrd" ]; then
+		if $(grep -q "^image.*" "${SYSCONFIG_FILE}"); then
+			if $(grep -q "^initrd.*" "${SYSCONFIG_FILE}"); then
+				sed -i '/^image.*/d' "${SYSCONFIG_FILE}"
+			else
+				sed -i 's|^image.*|initrd = "'${initrd_file}'"|g' "${SYSCONFIG_FILE}"
+			fi
+		fi
+	else
+		if $(grep -q "^initrd.*" "${SYSCONFIG_FILE}"); then
+			if $(grep -q "^image.*" "${SYSCONFIG_FILE}"); then
+				sed -i '/^initrd.*/d' "${SYSCONFIG_FILE}"
+			else
+				sed -i 's|^initrd.*|image = "'${image_file}'"|g' "${SYSCONFIG_FILE}"
+			fi
+		fi
+	fi
+
+	# enable debug
+	sed -i -e 's/^#\(enable_debug\).*=.*$/\1 = true/g' \
+		-e 's/^kernel_params = "\(.*\)"/kernel_params = "\1 agent.log=debug"/g' \
+		"${SYSCONFIG_FILE}"
+}
+
+run_test() {
+	setup_configuration_file "$1" "$2" "$3" "$4"
+
+	sudo -E kubectl apply -f "${SCRIPT_DIR}/runtimeclass_workloads/vfio.yaml"
+
+	pod_name=vfio
+	sudo -E kubectl wait --for=condition=Ready pod "${pod_name}"
+
+	net_interfaces=$(sudo -E kubectl exec -ti "${pod_name}" ip a | egrep -o "eth[0-9]+:" | wc -l)
```

Comment:
Are the if names guaranteed to use `eth` these days? I see `eth1` in the `resourceList`, but what is ensuring these "old" names are being used?